### PR TITLE
support for openidc on rhel and ubuntu

### PR DESCRIPTION
**Pull request by narengan**
Adding support for OpenID Connect federation to RHEL. Changed the code base to support both ubuntu and RHEL federation integration in Ursula.

### CODE REVIEW

**Comment by lihkin213** [comment]
retest

**Comment by lihkin213** [comment]
Please add a description.

**Comment by lihkin213** [comment]
ok to test

**Comment by epwn** [comment]
retest

**Comment by lihkin213** [comment]
@narengan please add a description to the PR

File: roles/keystone/templates/etc/apache2/sites-available/keystone.conf [hunk 1]
```diff
@@ -32,7 +32,11 @@ Listen {{ endpoints.keystone.port.backend_api }}
 {% macro setup_oidc() -%}
     OIDCClaimPrefix "OIDC_CLAIM_"
     OIDCClaimDelimiter ";"
+    {% if ursula_os == 'ubuntu' %}
```
Comment (nirajdp76) [inline roles/keystone/templates/etc/apache2/sites-available/keystone.conf]: seems like we can remove this if else condition and leverage - `/etc/{{ openstack_meta.apache[ursula_os].package_name }}/openidc/metadata`

**Comment by nirajdp76** [comment]
couple outstanding comments that needs to be addressed

File: roles/keystone/templates/etc/apache2/sites-available/keystone.conf (hunk 1 above)
Comment (narengan) [inline roles/keystone/templates/etc/apache2/sites-available/keystone.conf]: Ill make that change now

**Comment by narengan** [comment]
retest

**Pull request merged**